### DISCUSSION TITLE
How to change port

### DISCUSSION THREAD

**Discussion by lucaverardi** [agronholm/apscheduler#693] [Q&A]
Hi guys,
I have a beautiful scheduler.

```
class RegisterSchedule:
    def __init__(self, scheduler):
        self.__scheduler = scheduler
        self.__scheduler.add_job(updater.call_8_00, trigger='cron', hour='8', minute='1')
        self.__scheduler.add_job(updater.call_20_30, trigger='cron', hour='20', minute='30')
```

and a beautiful check if started

```
            try:
                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
                sock.bind(("127.0.0.1", 47200))
            except socket.error:
                logger.info("Do nothing")
            else:
                logger.info("Starting")
                scheduler.start()
```

If i have two or more project in same server how can i change port 47200?

Thank you

**Comment by agronholm** [comment]
I don't think this is directly related to APScheduler. But can't you take the port number as an environment variable, or command line argument?

**Comment by lucaverardi** [comment]
The two projects I mentioned earlier are pyramid projects that run on ports 6542 and 6543 and start with supervisord, but your answer begs another question. 
Port 47200 is a part of the apscheduler service?

I ask this question because I had a similar problem to what is written in this topic
https://stackoverflow.com/questions/16053364/make-sure-only-one-worker-launches-the-apscheduler-event-in-a-pyramid-web-app-ru
and solution of Paolo was good for me. 

This solution is still valid if I have two pyramid projects on the same server?

Thanks

**Reply by agronholm** [reply]
APScheduler does not bind any ports, which made me wonder why you're asking this here.

**Reply by lucaverardi** [reply]
ok i was completely wrong. thanks for all
